core/src/com/unciv/logic/city/CityResources.kt

```diff
@@ -67,6 +68,23 @@ object CityResources {
                 (unique.params[0].toFloat() * resourceModifer[resource.name]!!).toInt()
             )
         }
+
+        // StatPercentFromObjectToResource - Example: "[50]% of [Culture] from every [tileFilter/buildingFilter] in the city added to [Iron]"
+        for (unique in city.getMatchingUniques(UniqueType.StatPercentFromObjectToResource, city.state, false)) {
+            val resource = city.getRuleset().tileResources[unique.params[3]] ?: continue
+            val stat = Stat.safeValueOf(unique.params[1]) ?: continue
+            val percent = unique.params[0].toFloat() / 100f
+            val buildings = city.cityConstructions.getBuiltBuildings()
+                .filter { it.isStatRelated(stat) && it.matchesFilter(unique.params[2], city.state) }
+            for (building in buildings) {
+                val statValue = building.getStats(city)[stat] ?: 0f
+                val amount = (statValue * percent * resourceModifer[resource.name]!!).toInt()
```

Comment:
resourceModifer[resource.name]!! is constant across all buildings, should be taken out of the for loop